lib/metadata/index.js

```diff
@@ -158,11 +170,26 @@ Metadata.prototype.refreshKeyspace = function (name, callback) {
 
 /**
  * Gets the metadata information of all the keyspaces and updates the internal state of the driver.
+ * <p>
+ *   If a <code>callback</code> is provided, the callback is invoked when the keyspace metadata refresh completes.
+ *   Otherwise, it returns a <code>Promise</code>.
+ * </p>
  * @param {Boolean} [waitReconnect] Determines if it should wait for reconnection in case the control connection is not
  * connected at the moment. Default: true.
  * @param {Function} [callback] Optional callback.
  */
 Metadata.prototype.refreshKeyspaces = function (waitReconnect, callback) {
+  return utils.promiseWrapper.call(this, this.options, callback, true, function handler(cb) {
+    this._refreshKeyspacesCb(waitReconnect, cb);
+  });
+};
+
+/**
+ * @param {Boolean|Function} waitReconnect
+ * @param {Function} callback
+ * @private
+ */
+Metadata.prototype._refreshKeyspacesCb = function (waitReconnect, callback) {
```

Comment:
One problem here is the logic at 195,  if someone simply calls `refreshKeyspaces()`, `waitReconnect` will be undefined, and this would set callback = undefined, and then set it to noop, so the promise never completes.

In the callback case, things work fine.  I think maybe we need to adjust the logic in refreshKeyspaces to check in the same way.